module/VuFind/src/VuFind/Controller/AbstractBase.php

```diff
@@ -740,16 +747,44 @@ protected function normalizeUrlForComparison($url)
         return trim(end($parts), '/');
     }
 
+    /**
+     * Checks if a followup url is set
+     *
+     * @return bool
+     */
+    protected function hasFollowupUrl()
+    {
+        return null !== $this->followup()->retrieve('url');
+    }
+
     /**
      * Retrieve a referer to keep post-login redirect pointing
      * to an appropriate location.
      * Unset the followup before returning.
      *
+     * @param bool $checkRedirect Whether the query should be checked for param 'redirect'
+     *
      * @return string
      */
-    protected function getFollowupUrl()
-    {
-        return $this->followup()->retrieve('url', '');
+    protected function getAndClearFollowupUrl($checkRedirect = false)
+    {
+        if ($url = $this->followup()->retrieveAndClear('url', '')) {
```

Comment:
Is there a reason to default to `''` here instead of `null`? I would expect both to have the same falsy behavior for the purposes of this if statement.

Reply:
On a separate topic from the race condition discussed below, please let me know your thoughts on this item -- once it's resolved, I think we can merge this and move the rest of the discussion to a JIRA ticket or new PR.